src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectRpm.hpp

```diff
@@ -39,140 +40,226 @@ class VersionObjectRpm final : public IVersionObject
+     * involves several steps to accurately compare version strings that may include alphanumeric characters and tildes
+     * (~).
      *
-     * @see rpmvercmp.c
+     * The process begins by checking for string equality. If both version strings are identical, they are considered
+     * equal. If not, each string is converted into a list of characters for detailed comparison.
      *
-     * @param rRpmVersion The second RPM version.
+     * The comparison involves the following steps:
+     * - First, any leading characters that are neither alphanumeric nor tildes (~) are trimmed from both character
+     * lists.
+     * - Next, leading tildes (~) are removed. A tilde at the beginning of a version indicates that the version should
+     *   be considered older than versions without a tilde, even if numerically larger. Therefore, if one version starts
+     *   with a tilde and the other does not, the version without the tilde is considered newer.
+     * - The comparison then proceeds by popping and comparing consecutive blocks of digits or letters from the start of
+     *   each list. If a difference is found, the comparison result of these blocks is returned immediately.
+     * - If all characters are compared without finding a difference, the comparison concludes by examining the lengths
+     *   of any remaining characters in the lists. A version with remaining characters is considered newer, unless those
+     *   characters begin with a tilde, in which case it is considered older. This step accounts for scenarios where one
+     *   version string has been entirely consumed, leaving characters in the other, such as when comparing "1.05b" to
+     * "1.05".
      *
-     * @retval 0 If a and b are equal.
-     * @retval <0 If a is smaller than b.
-     * @retval >0 If a is greater than b.
+     * @param leftVer The left-hand side version string to compare.
+     * @param rightVer The right-hand side version string to compare.
+     * @return Returns LEFT_IS_NEWER if `leftVer` is considered newer, RIGHT_IS_NEWER if `rightVer` is newer, or
+     *   LEFT_EQ_RIGHT if both versions are considered equal.
      */
-    int compareRpmVersion(const std::string& rRpmVersion) const
+
+    int rpmvercmp(const char* leftVer, const char* rightVer) const
     {
         /* easy comparison to see if versions are identical */
-        if (!m_rpmVersion.compare(rRpmVersion))
-        {
+        if (std::strcmp(leftVer, rightVer) == 0)
             return 0;
-        }
-        const auto& lhs = m_rpmVersion;
-        auto rhs = rRpmVersion; // NOLINT
-        char* lhsStrAux = const_cast<char*>(lhs.c_str());
-        char* rhsStrAux = const_cast<char*>(rhs.c_str());
-        char* auxOne = lhsStrAux;
-        char* auxTwo = rhsStrAux;
-        char auxChrOne, auxChrTwo;
+
+        char oldChrLeft, oldChrRight;
+        char leftBuffer[strlen(leftVer) + 1], rightBuffer[strlen(rightVer) + 1];
```

Comment:
This length calculation if we keep the std::string is not necessary and can be avoided.